Libraries/Network/XMLHttpRequest.js

```diff
@@ -242,10 +253,11 @@ class XMLHttpRequest extends EventTarget(...XHR_EVENTS) {
         break;
 
       case 'blob':
-        this._cachedResponse = new global.Blob(
-          [base64.toByteArray(this._response).buffer],
-          {type: this.getResponseHeader('content-type') || ''}
-        );
+        if (typeof this._response === 'object' && this._response) {
+          this._cachedResponse = BlobManager.createFromOptions(this._response);
+        } else {
+          throw new Error('Invalid response for blob');
```

Comment:
Maybe also print out any useful debug information about `this._response`? Like its type?